### ISSUE TITLE
False Positive: Switches that are already exhaustive or intentionally omit `default` under the benchmark are still forced to add one.

### ISSUE THREAD

**Issue by Carlson-JLQ** [checkstyle/checkstyle#19389]
# False Positive: Switches that are already exhaustive or intentionally omit `default` under the benchmark are still forced to add one.

Version
unknown in repository metadata

## Checker
- Checker id: `MissingSwitchDefaultCheck`
- Checker description: Detects switch statements that lack a default case label and are not switch expressions, pattern case labels, or null case labels.

## Description of the false positive
`diff/output` marks 2 false positives for `MissingSwitchDefaultCheck`. Across these samples, the checker tends to fire on the outer syntax while missing code-level cues that make the pattern harmless, exempted, or at least much weaker than a real bug. The recurring themes are switch control flow.

### `NegCase2.java`
Source: `output/codeql/MissingDefaultInSwitch/src/main/java/scensct/core/neg/NegCase2.java`

The example is built around a `switch` whose case-group layout is the whole point of the test. A human reader would likely classify this as an allowed or intentionally structured pattern rather than a bug. That matches the benchmark's stated intent: A switch statement where the expression type is an enum type should not be flagged as missing default case.

```java
// A switch statement where the expression type is an enum type should not be flagged as missing default case.
package scensct.core.neg;

public class NegCase2 {
    // Define a minimal enum required for the scenario
    enum Color { RED, GREEN, BLUE }
    
    public static void main(String[] args) {
        Color color = Color.GREEN;
        // Enum type switch, default case not required for this checker's negative scenario
        switch (color) {
            case RED:
                System.out.println("Red");
                break;
            case GREEN:
                System.out.println("Green");
                break;
            case BLUE:
                System.out.println("Blue");
                break;
        }
    }
}
```

### `NegCase4.java`
Source: `output/errorprone/MissingDefault/src/main/java/scensct/core/neg/NegCase4.java`

The example is built around a `switch` whose case-group layout is the whole point of the test. A human reader would likely classify this as an allowed or intentionally structured pattern rather than a bug. That matches the benchmark's stated intent: A switch statement on enum type without default case should not be flagged as missing default when not exhaustive.

```java
// A switch statement on enum type without default case should not be flagged as missing default when not exhaustive.
package scensct.core.neg;

public class NegCase4 {
    enum Direction { NORTH, SOUTH }
    public static void main(String[] args) {
        Direction d = Direction.NORTH;
        switch (d) { // Enum type, no default case
            case NORTH:
                System.out.println("north");
                break;
            case SOUTH:
                System.out.println("south");
                break;
        }
    }
}
```

## Cause analysis
The rule is documented as follows: Detects switch statements that lack a default case label and are not switch expressions, pattern case labels, or null case labels.

In this set the problem shows up around switch control flow. The implementation appears to lean too heavily on surface indicators and not enough on what the code actually does.

That is why these reports read like over-approximation. The code often retains a suspicious signature or control-flow skeleton, but once you read the body, the benchmark's claimed exemption is visible in the source itself.

For `MissingSwitchDefaultCheck`, the practical improvement would be to model those source-level exemptions more explicitly instead of reporting every syntactic near-match.

## References
None known.

**Comment by romani** [comment]
please strictly follow issue description
